lib/hooks/typescript-compilation.ts

```diff
@@ -4,15 +4,15 @@ import fiberBootstrap = require("./../common/fiber-bootstrap");
 import { TARGET_FRAMEWORK_IDENTIFIERS } from "../common/constants";
 
 fiberBootstrap.run(() => {
-	$injector.require("typeScriptCompilationService", "./common/services/typescript-compilation-service");
 	let project: Project.IProject = $injector.resolve("project");
 	if (!project.projectData) {
 		return;
 	}
 
-	let typeScriptFiles = project.getTypeScriptFiles().wait();
+	let typeScriptService: ITypeScriptService = $injector.resolve("typeScriptService");
+	let typeScriptFiles = typeScriptService.getTypeScriptFiles(project.getProjectDir().wait()).wait();
+
 	if (typeScriptFiles.typeScriptFiles.length > typeScriptFiles.definitionFiles.length) { // We need this check because some of non-typescript templates(for example KendoUI.Strip) contain typescript definition files
-		let typeScriptCompilationService: ITypeScriptCompilationService = $injector.resolve("typeScriptCompilationService");
 		let $fs: IFileSystem = $injector.resolve("fs");
 		let pathToTsConfig = path.join(project.getProjectDir().wait(), "tsconfig.json");
```

Comment:
It appears the string `"tsconfig.json"` is used here too. You can use a constant should you chose to extract one in https://github.com/telerik/mobile-cli-lib/pull/776